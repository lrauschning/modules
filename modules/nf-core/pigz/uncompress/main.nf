process PIGZ_UNCOMPRESS {
    tag '$file'
    label 'process_low'

    conda "conda-forge::pigz"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/YOUR-TOOL-HERE': //TODO add when multicontainer has been built
        'biocontainers/YOUR-TOOL-HERE' }"

    input:
    path zip

    output:
    path $zip.dropRight(3), emit: file
    path "versions.yml"   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    unpigz \\
        -p $task.cpus \\
        $args \\
        $zip

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\w*//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''

    """
    touch ${zip.dropRight(3)}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pigz: \$(echo \$(pigz --version 2>&1) | sed 's/^.*pigz\w*//' ))
    END_VERSIONS
    """
}
