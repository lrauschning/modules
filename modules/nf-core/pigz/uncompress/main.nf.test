nextflow_process {

    name "Test Process PIGZ_UNCOMPRESS"
    script "modules/nf-core/pigz/uncompress/main.nf"
    process "PIGZ_UNCOMPRESS"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                // input[0] = file("test-file.txt")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
