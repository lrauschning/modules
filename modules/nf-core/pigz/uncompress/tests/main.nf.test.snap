{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                ],
                "1": [
                    "versions.yml:md5,7c265953bddf5eaf230bcd49fd349af9"
                ],
                "file": [
                    "test_1.fastq:md5,4161df271f9bfcd25d5845a1e220dbec"
                ],
                "versions": [
                    "versions.yml:md5,7c265953bddf5eaf230bcd49fd349af9"
                ]
            }
        ],
        "timestamp": "2023-12-11T19:09:50.385774055"
    }
}